connectivity/tests/encryption.go

```diff
@@ -0,0 +1,179 @@
+	} else {
+		cmd := []string{"/bin/sh", "-c",
+			fmt.Sprintf("ip -o r g %s | grep -oE 'dev [^ ]*' | cut -d' ' -f2", server.Pod.Status.PodIP)}
+		t.Debugf("Running %s", strings.Join(cmd, " "))
+		dev, err := clientHost.K8sClient.ExecInPod(ctx, clientHost.Pod.Namespace,
+			clientHost.Pod.Name, "", cmd)
+		if err != nil {
+			t.Fatalf("Failed to get IP route: %s", err)
+		}
+		iface = strings.TrimRight(dev.String(), "\n\r")
+	}
+
+	bgStdout := &safeBuffer{}
+	bgStderr := &safeBuffer{}
+	bgExited := make(chan struct{})
+	// Start kubectl exec in bg (=goroutine)
+	go func() {
+		// Run tcpdump with -w instead of directly printing captured pkts. This
+		// is to avoid a race after sending ^C (triggered by bgCancel()) which
+		// might terminate the tcpdump process before it gets a chance to dump
+		// its captures.
+		cmd := []string{
+			"tcpdump", "-i", iface, "--immediate-mode", "-w", "/tmp/foo.pcap",
+			// Unfortunately, we cannot use "host %s and host %s" filter here,
+			// as IPsec recirculates replies to the iface netdev, which would
+			// make tcpdump to capture the pkts (false positive)
+			fmt.Sprintf("src host %s and dst host %s", client.Pod.Status.PodIP, server.Pod.Status.PodIP),
+			// Only one pkt is enough, as we don't expect any unencrypted pkt
+			// to be captured
+			"-c", "1"}
+		t.Debugf("Running in bg: %s", strings.Join(cmd, " "))
+		err := clientHost.K8sClient.ExecInPodWithWriters(bgCtx, clientHost.Pod.Namespace,
+			clientHost.Pod.Name, "", cmd, bgStdout, bgStderr)
+		if err != nil {
+			t.Fatalf("Failed to execute tcpdump: %s", err)
+		}
+		close(bgExited)
+	}()
+
+	// Wait until tcpdump is ready to capture pkts
+	timeout := time.After(5 * time.Second)
+	for found := false; !found; {
+		select {
+		case <-timeout:
+			t.Fatalf("Failed to wait for tcpdump to be ready")
+		default:
+			line, err := bgStdout.ReadString('\n')
+			if err != nil && !errors.Is(err, io.EOF) {
+				t.Fatalf("Failed to read kubectl exec's stdout: %s", err)
+			}
+			if strings.Contains(line, fmt.Sprintf("listening on %s", iface)) {
+				found = true
+				break
+			}
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
+
+	// Curl the server from the client to generate some traffic
+	t.NewAction(s, "curl", &client, server).Run(func(a *check.Action) {
+		a.ExecInPod(ctx, curl(server))
+	})
+
+	// Wait until tcpdump has exited
+	bgCancel()
+	<-bgExited
+
+	// Redirect stderr to /dev/null, as tcpdump logs to stderr, and ExecInPod
+	// will return an error if any char is written to stderr. Anyway, the count
+	// is written to stdout.
+	cmd := []string{"/bin/sh", "-c", "tcpdump -r /tmp/foo.pcap --count 2>/dev/null"}
+	count, err := clientHost.K8sClient.ExecInPod(ctx, clientHost.Pod.Namespace, clientHost.Pod.Name, "", cmd)
+	if err != nil {
+		t.Fatalf("Failed to retrieve tcpdump pkt count: %s", err)
+	}
+	if !strings.HasPrefix(count.String(), "0 packets") {
+		t.Fatalf("Captured unencrypted pkt (count=%s)", count.String())
```

Comment:
I can already predict the count here. It's going to be 1 :grin:

Reply:
It's also a test for tcpdump ;-)